src/SB/Game/zScene.cpp

```diff
@@ -3512,8 +3498,8 @@ void zSceneEnableScreenAdj(U32 enable)
 
 void zSceneSetOldScreenAdj()
 {
-    oldOffsetx = offsetx;
-    oldOffsety = offsety;
+    *(volatile F32*)(&oldOffsetx) = offsetx;
+    *(volatile F32*)(&oldOffsety) = offsety;
```

Comment:
Rather than casting the variables here like this, it would be preferable to leave the simpler code intact and add `volatile` to the "old" variable declarations. From what I can see, no other code makes use of these values.